EmbeddingWithAmazonLabelDocument.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Newtonsoft.Json;
+
+namespace VectorIndexScenarioSuite
+{
+    internal class EmbeddingWithAmazonLabelDocument : EmbeddingDocumentBase
+    {
+        [JsonProperty(PropertyName = "brand")]
+        private string Brand { get; }
+
+        //ratting 
+        [JsonProperty(PropertyName = "rating")]
+        private string Rating { get; }
+
+        // category
+        [JsonProperty(PropertyName = "category")]
+        private string[] Category { get; }
+
+        public EmbeddingWithAmazonLabelDocument(string id, float[] embedding, string label)
+            : base(id, embedding) // Call the base class constructor
+        {
+            // Parse the label string into brand, rating, and category
+            var labelJson = ParseAmazonLabelToJson(label);
+            this.Brand = labelJson["brand"]?.ToString() ?? string.Empty;
+            this.Rating = labelJson["rating"]?.ToString() ?? string.Empty;
+            this.Category = ((List<string>?)labelJson["category"])?.ToArray() ?? Array.Empty<string>();
+        }
+
+        public static Dictionary<string, object> ParseAmazonLabelToJson(string line)

```

Comment:
Please don't mix file format passing with document format. 
Please add this to 'BigANNBinaryFormat' if this label format is specific to BigANN?